main.go

```diff
@@ -2,71 +2,132 @@ package main
 
 import (
 	"bytes"
+	_ "embed"
 	"flag"
 	"fmt"
 	"log"
 	"os"
 	"text/template"
 
 	"github.com/nicheinc/mock/iface"
+	"golang.org/x/tools/go/packages"
 	"golang.org/x/tools/imports"
 )
 
+//go:embed template.tmpl
+var tmpl string
+
+const helpMessage = `Usage: %s [options] [interface]
+
+When the positional interface argument is omitted, all interfaces in the search
+directory annotated with a "go:mock [output file]" directive will be mocked and
+output to stdout or, with the -w option, written to files. If a go:mock
+directive in a file called example.go doesn't specify an output file, the
+default output file will be example_mock.go.
+
+When an interface name is provided as a positional argument after all other
+flags, only that interface will be mocked. The -o option requires an interface
+argument, and the -w option is incompatible with an interface argument.
+
+Options:
+`
+
+type config struct {
+	dir        string
+	outputFile string
+	write      bool
+}
+
 func main() {
-	var (
-		dir     = flag.String("d", ".", "Directory to search for interface in")
-		outFile = flag.String("o", "", "Output file (default stdout)")
-	)
+	var config config
+	flag.StringVar(&config.dir, "d", ".", "Directory to search for interfaces in")
+	flag.StringVar(&config.outputFile, "o", "", "Output file (default stdout)")
+	flag.BoolVar(&config.write, "w", false, "Write mocks to files rather than stdout")
+
 	flag.Usage = func() {
-		fmt.Fprintf(flag.CommandLine.Output(), "Usage: %s [options] interface\n", os.Args[0])
-		fmt.Fprintf(flag.CommandLine.Output(), "Options:\n")
+		fmt.Fprintf(flag.CommandLine.Output(), helpMessage, os.Args[0])
 		flag.PrintDefaults()
 	}
 	flag.Parse()
 
-	// First argument is interface name
-	args := flag.Args()
-	if len(args) < 1 {
-		log.Fatal("Not enough args")
+	// Load package info.
+	pkgs, packageErr := packages.Load(&packages.Config{Mode: packages.LoadSyntax}, config.dir)
+	if packageErr != nil {
+		log.Fatalf(`Error loading package information: %s`, packageErr)
 	}
-	ifaceName := args[0]
-
-	// Parse the package and get info about the interface
-	iface, err := iface.GetInterface(*dir, ifaceName)
-	if err != nil {
-		log.Fatalf("Error getting interface information: %s", err)
+	if len(pkgs) < 1 {
+		log.Fatalf(`No packages found in %s`, config.dir)
 	}
```

Comment:
I pulled this out of `GetInterface` since it's now the common first step of both single-interface and multi-interface mode.